junit-jupiter-api/src/main/kotlin/org/junit/jupiter/api/Assertions.kt

```diff
@@ -86,6 +102,201 @@ fun assertAll(vararg executables: () -> Unit) =
+fun assertNull(actual: Any?, messageSupplier: () -> String) {
+    contract {
+        returns() implies (actual == null)
+
+        callsInPlace(messageSupplier, AT_MOST_ONCE)
+    }
+
+    Assertions.assertNull(actual, messageSupplier)
+}
+
+/**
+ * Example usage:
+ * ```kotlin
+ * val string: String? = ...
+ *
+ * assertNotNull(string)
+ *
+ * // compiler smart casts nullableString to a non-nullable object
+ * assertTrue(string.isNotEmpty())
+ * ```
+ * @see Assertions.assertNotNull
+ */
+@API(since = "5.11", status = EXPERIMENTAL)
+fun assertNotNull(actual: Any?) {
+    contract {
+        returns() implies (actual != null)
+    }
+
+    Assertions.assertNotNull(actual)
+}
+
+/**
+ * Example usage:
+ * ```kotlin
+ * val string: String? = ...
+ *
+ * assertNotNull(string, "Should be non-nullable")
+ *
+ * // compiler smart casts nullableString to a non-nullable object
+ * assertTrue(string.isNotEmpty())
+ * ```
+ * @see Assertions.assertNotNull
+ */
+@API(since = "5.11", status = EXPERIMENTAL)
+fun assertNotNull(actual: Any?, message: String) {
+    contract {
+        returns() implies (actual != null)
+    }
+
+    Assertions.assertNotNull(actual, message)
+}
+
+/**
+ * Example usage:
+ * ```kotlin
+ * val string: String? = ...
+ *
+ * assertNotNull(string) { "Should be non-nullable" }
+ *
+ * // compiler smart casts nullableString to a non-nullable object
+ * assertTrue(string.isNotEmpty())
+ * ```
+ * @see Assertions.assertNotNull
+ */
+@API(since = "5.11", status = EXPERIMENTAL)
+fun assertNotNull(actual: Any?, messageSupplier: () -> String) {
+    contract {
+        returns() implies (actual != null)
+
+        callsInPlace(messageSupplier, AT_MOST_ONCE)
+    }
+
+    Assertions.assertNotNull(actual, messageSupplier)
+}
+
+/**
+ * Example usage:
+ * ```kotlin
+ * val string: Any = ...
+ *
+ * assertInstanceOf<String>(string)
+ *
+ * // compiler smart casts string to a String object
+ * assertTrue(string.isNotEmpty())
+ * ```
+ * @see Assertions.assertInstanceOf
+ */
+@API(since = "5.11", status = EXPERIMENTAL)
+inline fun <reified T : Any> assertInstanceOf(actual: Any?) {
+    contract {
+        returns() implies (actual is T)
+    }
+
+    Assertions.assertInstanceOf(T::class.java, actual)
+}
+
+/**
+ * Example usage:
+ * ```kotlin
+ * val string: Any = ...
+ *
+ * assertInstanceOf<String>(string, "Should be a String")
+ *
+ * // compiler smart casts string to a String object
+ * assertTrue(string.isNotEmpty())
+ * ```
+ * @see Assertions.assertInstanceOf
+ */
+@API(since = "5.11", status = EXPERIMENTAL)
+inline fun <reified T : Any> assertInstanceOf(actual: Any?, message: String) {
+    contract {
+        returns() implies (actual is T)
+    }
+
+    Assertions.assertInstanceOf(T::class.java, actual, message)
+}
+
+/**
+ * Example usage:
+ * ```kotlin
+ * val string: Any = ...
+ *
+ * assertInstanceOf<String>(string) { "Should be a String" }
+ *
+ * // compiler smart casts string to a String object
```

Comment:
Would marking variables with backticks make this a bit more easy to read?
```suggestion
 * // compiler smart casts `string` to a String object
```
or maybe a different variable name?

(applies to all docs examples)

Reply:
The examples in the docs have been revised and variable names have been slightly adjusted